### DISCUSSION TITLE
Break `api.py` file into several for readability

### DISCUSSION THREAD

**Discussion by ihelmer07** [vitalik/django-ninja#471] [Q&A] [answered]
I need to create the CRUD api endpoints for over 20 models.
As a best practice for my team, we break the models and views by individual model.

e.g. `Program` object would mean we create a `models/program.py` and a `views/program_views.py`

How would I create a multiple api files while still using the single api object?

Per the docs - one creates  an `api.py` file
```python
from ninja import NinjaAPI

api = NinjaAPI()


@api.get("/hello")
def hello(request):
    return "Hello world"
```

and in `urls.py`:
```python
from django.contrib import admin
from django.urls import path
from .api import api

urlpatterns = [
    path("admin/", admin.site.urls),
    path("api/", api.urls),
]
```

I want to turn `api.py` into a folder (with `__init__.py`) and have:
`api/api_class_a.py`
```python
from ninja import NinjaAPI
api = NinjaAPI()


@api.get("/hello")
def hello(request):
    return "Hello world"

```

AND

`api/api_class_b.py`
```python
@api.get("/hi-you-there")
def hello_class_b(request):
    return "Hi You Are In Class B"
```

currently if I try that - the 'hello' endpoint is showing but the hello_class_b is not.

**Comment by SmileyChris** [comment, marked as answer]
That should work fine, as long as each module actually gets imported before `api.urls` is called. Otherwise the decorators adding your functions to the api will never be executed!

You could either do these imports on the urlconf (`import .api.api_class_b`)

Or you could import every split module class from `api/__init__.py`. If you do that though, you'll want to put the actual `api = NinjaAPI()` in a child module rather than in the __init__ root itself to avoid a circular import.

**Comment by ihelmer07** [comment]
To expand for prosperity:

This works:

in `__init__.py`
```python 

from . import api_class_b.py
from .api_class_a import api # django_ninja api object
```
in `api_class_b.py`
```python
from .api_class_a import api

@api.get("/hi-you-there")
def hello_class_b(request):
    return "Hi You Are In Class B"
```

in `api_class_a.py`
```python 
from ninja import NinjaAPI
api = NinjaAPI()


@api.get("/hello")
def hello(request):
    return "Hello **world"**
```
